src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/interior_data_cache_impl.cc

```diff
@@ -60,11 +68,54 @@ smart_objects::SmartObject MergeModuleData(
   for (; it != data2.map_end(); ++it) {
     const std::string& key = it->first;
     const smart_objects::SmartObject& value = it->second;
-    result[key] = value;
+    // Merge maps and arrays with `id` param included, replace other types
+    if (result.keyExists(key) && value.getType() == result[key].getType()) {
+      if (value.getType() == smart_objects::SmartType::SmartType_Map) {
+        result[key] = MergeModuleData(result[key], value);
+      } else if (value.getType() == smart_objects::SmartType::SmartType_Array) {
+        result[key] = MergeArray(result[key], value);
+      } else {
+        result[key] = value;
+      }
+    } else {
+      result[key] = value;
+    }
   }
   return result;
 }
 
+smart_objects::SmartObject MergeArray(const smart_objects::SmartObject& data1,
+                                      const smart_objects::SmartObject& data2) {
+  // Merge data only in the case where each value in the array is an Object with
+  // an ID included, otherwise replace
+  if (!data2.empty() &&
```

Comment:
@jacobkeeler it is complicated to read 3 conditions in one if condition

Reply:
```cpp
DCHECK(data2.getElement(0).getType() !=
          smart_objects::SmartType::SmartType_Map); 
```
This function called from  MergeModuleData, and if this is map, there is mistake in MergeModuleData function, isn't it?

Reply:
```cpp
if (data2.empty() || data2.getElement(0).keyExists(application_manager::strings::id)) {
  return data2;
}
```

Reply:
I would move this lambda out of loop, and give easy readable name
```cpp
auto result_contains = [&result_array](const smart_objects::SmartObject& id) {
   auto result_it = std::find_if(result_array->begin(), result_array->end(),
                       [&id](smart_objects::SmartObject& obj) -> bool {
                         return obj[application_manager::strings::id] == id;
                       });
}
```

Reply:
> ```c++
> DCHECK(data2.getElement(0).getType() !=
>           smart_objects::SmartType::SmartType_Map); 
> ```
> 
> This function called from MergeModuleData, and if this is map, there is mistake in MergeModuleData function, isn't it?

This is checking the type of the first element, not the array itself.